code/lib/cli/src/js-package-manager/NPMProxy.ts

```diff
@@ -60,6 +61,27 @@ export class NPMProxy extends JsPackageManager {
     return this.uninstallArgs;
   }
 
+  setRegistryURL(url: string) {
+    if (url) {
+      this.executeCommand('npm', ['config', 'set', 'registry', url]);
+    } else {
+      this.executeCommand('npm', ['config', 'delete', 'registry']);
+    }
+  }
+
+  getRegistryURL() {
+    return this.executeCommand('npm', ['config', 'get', 'registry']).trim();
```

Comment:
Why is the logic here different than the yarn one? When it comes to handling undefined return value